src/serviceworker.js

```diff
@@ -0,0 +1,140 @@
+                  } failed with status ${
+                    response.statusText}`
+                )
+              }
+
+              // Use the original URL without the cache-busting parameter
+              // as the key for cache.put().
+              return cache.put(urlToPrefetch, response)
+            })
+            .catch(error => {
+              console.error(
+                `Not caching ${urlToPrefetch} due to ${error}`
+              )
+            })
+        })
+
+        return Promise.all(cachePromises).then(() => {
+          console.log("Pre-fetching complete.")
+        })
+      })
+      .catch(error => {
+        console.error("Pre-fetching failed:", error)
+      })
+  )
+})
+
+self.addEventListener("activate", event => {
+  // Delete all caches that aren't named in CURRENT_CACHES.
+  // While there is only one cache in this example,
+  // the same logic will handle the case where
+  // there are multiple versioned caches.
+  const expectedCacheNames = Object.keys(CURRENT_CACHES).map(key => {
+    return CURRENT_CACHES[key]
+  })
+
+  event.waitUntil(
+    caches.keys().then(cacheNames => {
+      return Promise.all(
+        cacheNames.map(cacheName => {
+          if (expectedCacheNames.indexOf(cacheName) === -1) {
+            // If this cache name isn't present in the array of "expected"
+            // cache names, delete it.
+            console.log("Deleting out of date cache:", cacheName)
+            return caches.delete(cacheName)
+          } else {
+            return false
+          }
+        })
+      )
+    })
+  )
+})
+
+self.addEventListener("fetch", event => {
+  console.log("Handling fetch event for", event.request.url)
+
+  event.respondWith(
+    // caches.match() will look for a cache entry in all of the caches available
+    // to the service worker. It's an alternative to first opening
+    // a specific named cache and then matching on that.
+    caches.match(event.request).then(response => {
+      if (response) {
+        console.log("Found response in cache:", response)
+
+        return response
+      }
+
+      console.log("No response found in cache. About to fetch from network...")
+
+      // event.request will always have the proper mode set
+      // ('cors, 'no-cors', etc.) so we don't have to hardcode
+      // 'no-cors' like we do when fetch()ing in the install handler.
+      return fetch(event.request)
+        // eslint-disable-next-line no-shadow
+        .then(response => {
+          console.log("Response from network is:", response)
+
+          return response
+        })
+        .catch(error => {
+          // This catch will handle exceptions thrown from the fetch operation.
+          // Note that a HTTP error response will NOT trigger an exception.
+          // It will return a normal response object that has the error code set
+          console.error("Fetching failed:", error)
+
+          throw error
+        })
+    })
+  )
+})
```

Comment:
This looks super-generic. Is there maybe an out-of-the-box solution which we can configure and use?